streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -80,22 +80,30 @@ public void shouldLockTaskStateDirectory() throws Exception {
 
         directory.lock(taskId, 0);
 
-        final FileChannel channel = FileChannel.open(new File(taskDirectory, StateDirectory.LOCK_FILE_NAME).toPath(), StandardOpenOption.CREATE, StandardOpenOption.WRITE);
-        try {
+        try (
+            final FileChannel channel = FileChannel.open(
+                new File(taskDirectory, StateDirectory.LOCK_FILE_NAME).toPath(),
+                StandardOpenOption.CREATE, StandardOpenOption.WRITE)
+        ) {
             channel.tryLock();
             fail("shouldn't be able to lock already locked directory");
-        } catch (OverlappingFileLockException e) {
+        } catch (final OverlappingFileLockException e) {
             // pass
+        } finally {
+            directory.unlock(taskId);
```

Comment:
Hmmm... Not sure. If all test would only use `TaskId(0,0)` I would agree. However, different test use different `TaskId`s, thus, it also "messy" to keep track which one is used. The advantage of "locally isolated" test is, that you see at one shot what's going on -- each test create the `TaskId` and thus is responsible to unlock it. It does not make sense to me, to add the `TaskId` as global test vars.